src/pages/index.js

```diff
@@ -66,20 +108,85 @@ function renderCard(data) {
 /*-------------------------------------------------------------------------------*/
 
 function handleProfileFormSubmit(data) {
+  profileEditPopup.setLoading(true);
   userInfo.setUserInfo(data);
+  api.updateUserInfo(userInfo.getUserInfo())
+  .catch((err) => {
+    console.error(err);
+  })
+  .finally(() => profileEditPopup.setLoading(false));
   profileFormValidator.resetValidation();
   profileEditPopup.close();
 }
 
+function handleAvatarFormSubmit(data) {
+avatarEditPopup.setLoading(true);
+const avatar = data.link;
+userInfo.setAvatar({ avatar });
+api.updateAvatar(data.link)
+.catch((err) => {
+  console.error(err);
+})
+.finally(() => avatarEditPopup.setLoading(false));
+avatarFormValidator.resetValidation();
+avatarEditPopup.close();
+}
+
 function handleCardFormSubmit(data) {
-  renderCard(data, cardList);
+  cardAddPopup.setLoading(true);
+  api.addCard(data)
+  .then((card) => {
+  renderCard(card, cardList);
   cardFormValidator.resetValidation();
-  cardAddPopup.close();
+  cardAddPopup.close();})
+  .catch((err) => {
+    console.error(err);
+  })
+  .finally(() => cardAddPopup.setLoading(false));
 }
 
+
 function handleImageClick({ link, name }) {
   previewImagePopup.open({ link, name });
 }
+
+function handleCardDelete(cardId, card) {
+  console.log(cardId);
+  console.log(card);
+  deleteConfirmPopup.open();
+deleteConfirmPopup.setConfirmFunction( () => {
+  api.removeCard(cardId)
+  .catch((err) => {
+    console.error(err);
+  });
+  card.remove();
+  deleteConfirmPopup.close()
+});
+}
+
+function handleCardLike(card, likeStatus, cardId) {
+if(!likeStatus){
+  api.addCardLike(cardId).then((res) => {
+    console.log(res.isLiked);
+    card.setLiked(res.isLiked);
+  }
+)
+.catch((err) => {
```

Comment:
You need to format the markup (and code) to show **indentation** and to make the code more readable. You can do that with `prettier` or a combination `SHIFT + ALT + F`

**Please, correct it everywhere in the code**